rasa_nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -316,38 +306,41 @@ def _create_tf_embed_nn(self, x_in, is_training,
         return x
 
     def _create_tf_embed(self, a_in, b_in, is_training):
+        # type: (tf.Tensor, tf.Tensor, Bool) -> tf.Tensor
         """Create tf graph for training"""
 
         emb_a = self._create_tf_embed_nn(a_in, is_training,
-                                         self.num_hidden_layers_a,
-                                         self.hidden_layer_size_a,
+                                         self.hidden_layer_sizes['a'],
                                          name='a')
         emb_b = self._create_tf_embed_nn(b_in, is_training,
-                                         self.num_hidden_layers_b,
-                                         self.hidden_layer_size_b,
+                                         self.hidden_layer_sizes['b'],
                                          name='b')
         return emb_a, emb_b
 
     def _tf_sim(self, a, b):
-        """Define similarity"""
+        # type: (tf.Tensor, tf.Tensor) -> Tuple[tf.Tensor, tf.Tensor]
+        """Define similarity in two cases:
+            sim: between embedded words and embedded intent labels
+            sim_emb: between individual embedded intent labels only"""
 
         if self.similarity_type == 'cosine':
+            # normalize embedding vectors for cosine similarity
             a = tf.nn.l2_normalize(a, -1)
             b = tf.nn.l2_normalize(b, -1)
 
-        if self.similarity_type == 'cosine' or self.similarity_type == 'inner':
+        if self.similarity_type in ['cosine', 'inner']:
             sim = tf.reduce_sum(tf.expand_dims(a, 1) * b, -1)
-
-            # similarity between intent embeddings
             sim_emb = tf.reduce_sum(b[:, 0:1, :] * b[:, 1:, :], -1)
 
             return sim, sim_emb
+
         else:
             raise ValueError("Wrong similarity type {}, "
                              "should be 'cosine' or 'inner'"
                              "".format(self.similarity_type))
 
     def _tf_loss(self, sim, sim_emb):
+        # type: (tf.Tensor, tf.Tensor) -> tf.Tensor
         """Define loss"""
 
         if self.use_max_sim_neg:
```

Comment:
let's also refactor loss as in embedding policy, there is no need for `factors` tensor